### PR TITLE
[RLlib] Cleanup examples folder #10: Add `custom_rl_module.py` example script and matching RLModule example class (tiny CNN)..

### PR DESCRIPTION
**Pull request by sven1977**
Cleanup examples folder #10: Add `custom_rl_module.py` example script and matching RLModule example class (tiny CNN)..

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/core/rl_module/torch/torch_rl_module.py [hunk 1]
```diff
@@ -156,6 +132,20 @@ def _inference_only_get_state_hook(
         """
         pass
 
+    def _get_default_action_dist_class(self, what: str) -> Type[TorchDistribution]:
+        # The default implementation is to return TorchCategorical for Discrete action
+        # spaces and TorchDiagGaussian for Box action spaces. For all other spaces,
+        # raise a NotImplementedError
+        if isinstance(self.config.action_space, gym.spaces.Discrete):
```
Comment (simonsays1980) [inline rllib/core/rl_module/torch/torch_rl_module.py:139]: Why not using  `TorchMultiCategorical` and `TorchMultiDistribution` - things that get assembled inside of the `Catalog`?

File: rllib/env/single_agent_env_runner.py [hunk 2]
```diff
@@ -91,12 +91,9 @@ def __init__(self, config: AlgorithmConfig, **kwargs):
         try:
             module_spec: SingleAgentRLModuleSpec = self.config.rl_module_spec
             module_spec.observation_space = self._env_to_module.observation_space
-            # TODO (simon): The `gym.Wrapper` for `gym.vector.VectorEnv` should
```
Comment (simonsays1980) [inline rllib/env/single_agent_env_runner.py:94]: Great that this is gone now.

**Review by simonsays1980 — APPROVED** [review]
LGTM.

File: rllib/env/single_agent_env_runner.py (hunk 2 above)
Comment (sven1977) [reply rllib/env/single_agent_env_runner.py:94]: Yeah, it didn't seem to be a problem anymore (e.g. for PPO Pendulum, everything looks completely fine w/o any weird space errors on the Box actions). So I removed this comment.

File: rllib/core/rl_module/torch/torch_rl_module.py (hunk 1 above)
Comment (sven1977) [reply rllib/core/rl_module/torch/torch_rl_module.py:139]: Let's continue brainstorming how to simplify the general RLModule experience for the user ...

**Pull request merged**